src/parser/scanner_support.c

```diff
@@ -531,214 +638,424 @@ pt_get_hint (const char *text, PT_HINT hint_table[], PT_NODE * node)
+    {
+      in++;
+    }
+
+  if (*in == ')')
+    {
+      if (is_first_hit)
+	{
+	  if (IS_HINT_ON_TABLE (hint_table[hint_idx].hint))
+	    {
+	      /*
+	       * INDEX_SS() or INDEX_LS() means do not apply
+	       * hint, use special node to mark this.
+	       */
+	      PT_NODE *arg = parser_new_node (this_parser, PT_VALUE);
+	      if (arg)
+		{
+		  arg->type_enum = PT_TYPE_NULL;
+		}
+	      hint_table[hint_idx].arg_list = arg;
+	      hint_table[hint_idx].is_hit = true;
+	      *result_hint |= hint_table[hint_idx].hint;
+	    }
+	}
+
+      return in + 1;
+    }
+
+  sp = in;
+  while (*in)
+    {
+      if (*in == '"' || *in == '[' || *in == '`')
+	{
+	  delimeter = (*in == '[') ? ']' : *in;
+	  in++;
+	  while (*in)
+	    {
+	      if (*in == '\\' && *(in + 1) == delimeter)
+		{
+		  in++;
+		}
+	      else if (*in == delimeter)
+		{
+		  in++;
+		  break;
+		}
+	      in++;
+	    }
+	}
+
+      // IS_WHITE_SPACE (*in)
+      if (*in == '(')
+	{
+	  /* illegal hint expression */
+	  is_illegal_expression = true;
+	}
+      else if (*in == '.')
+	{
+	  if (!dot_ptr)
+	    {
+	      dot_ptr = in;	// set the position of the first dot
+	    }
+	}
+      else if (*in == ',')
+	{
+	  if (is_first_hit && !is_illegal_expression)
+	    {
+	      get_hint_args_func (sp, in, dot_ptr, &(hint_table[hint_idx]));
+	    }
+	  dot_ptr = 0x00;
+	  sp = in + 1;
+	}
+      else if (*in == ')')
+	{			// OK
+	  if (is_first_hit && !is_illegal_expression)
+	    {
+	      get_hint_args_func (sp, in, dot_ptr, &(hint_table[hint_idx]));
+	      hint_table[hint_idx].is_hit = true;
+	      *result_hint |= hint_table[hint_idx].hint;
+	    }
+
+	  return in + 1;
+	}
+
+      in++;
+    }
+
+  /* illegal hint expression */
+  if (is_first_hit)
+    {
+      parser_free_node (this_parser, hint_table[hint_idx].arg_list);
+      hint_table[hint_idx].arg_list = NULL;
+
+#if 1
+      // This code has been inserted to handle the same as the existing code.
+      // It responds to the following types of errors:  INDEX_SS( (idx)
+      if (IS_HINT_ON_TABLE (hint_table[hint_idx].hint))
+	{
+	  /*
+	   * INDEX_SS() or INDEX_LS() means do not apply
+	   * hint, use special node to mark this.
+	   */
+	  PT_NODE *arg = parser_new_node (this_parser, PT_VALUE);
+	  if (arg)
+	    {
+	      arg->type_enum = PT_TYPE_NULL;
+	    }
+	  hint_table[hint_idx].arg_list = arg;
+	  hint_table[hint_idx].is_hit = true;
+	  *result_hint |= hint_table[hint_idx].hint;
+	}
+#endif
+    }
+
+  return in;
 }
 
 /*
  * pt_check_hint () - search hint comment from hint table
  *   text(in): hint comment
  *   hint_table(in): hint info structure list
  *   result_hint(in): found result
- *   prev_is_white_char(in): flag indicates prev char
+ *   prev_is_white_char(in): -
  */
 void
-pt_check_hint (const char *text, PT_HINT hint_table[], PT_HINT_ENUM * result_hint, bool prev_is_white_char)
+pt_check_hint (const char *text, PT_HINT hint_table[], PT_HINT_ENUM * result_hint, bool)
```

Comment:
```suggestion
pt_check_hint (const char *text, PT_HINT hint_table[], PT_HINT_ENUM * result_hint, bool prev_is_white_char)
```
typo? or intended to remove the last argument?

Reply:
prev_is_white_char is a meaningless parameter, so it is omitted.
Let's remove the parameter itself.